Ref/RecvBuffApp/RecvBuffComponentImpl.cpp

```diff
@@ -34,16 +34,16 @@ namespace Ref {
         // reset deserialization of buffer
         buff.resetDeser();
         // deserialize packet ID
-        U32 id;
+        U32 id = 0;
         Fw::SerializeStatus stat = buff.deserialize(id);
         FW_ASSERT(stat == Fw::FW_SERIALIZE_OK,static_cast<NATIVE_INT_TYPE>(stat));
         // deserialize data
-        U8 testData[24];
+        U8 testData[24] = {0};
```

Comment:
@saba-ja what are the C++ standard requirements for this line?  I don't think this works with all compilers, yes?

Reply:
@LeStarch given that this is supported in C99 I think we generally count on C++ support for this.

Reply:
It did compile and run below, and the RPI compiler was the one I was worried about.

Reply:
...except this is the Ref app, so it didn't.  I'll check locally.

Reply:
@LeStarch ,

Short answer:
This only works if the value is `{0}` and I believe won't cause an issue in C++98 and higher.
Please advice if there is any better way to initialize this array.

Long answer:
`{0}` sets the first value of the array to `0` as specified by the user and the rest of the elements are zero initialized by the compiler. If the value is anything other than `0` only the first value will be set and the rest of the elements will be set to zero by the compiler.

I believe brace-enclosed initialization has been supported in C lang since C99
see: https://en.cppreference.com/w/c/language/array_initialization

It is supported in C++11 standard (see section 3.9.3 part 5 in page 74 of http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3690.pdf)

Did not find specification of it in C++98 but ran the following with the following and did not see any warnings:

```
g++ --std=c++98 -pedantic -Wall -Wextra -Wcast-align -Wdisabled-optimization -Winit-self -Werror test.cpp
```

test.cpp
```c++
#include <stdio.h>

int main() {
    
    int a[10] = {0};
    
   for(int i=0; i < 10; i++) {
    printf("%d ", a[i]);     
   }  
    return 0;
}
```
Output:
```
0 0 0 0 0 0 0 0 0 0
```

Reply:
Compiled on Lionaro ARM GCC 7.5 (gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf) with no warning.

Reply:
I just confirmed with the RPI tools too.  We can ship this, and initialize differently if it raises issues somewhere.